app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,31 @@
+class ProjectsController < ApplicationController
+  @@MAX_VOTES = 3
+
+  def create
+    @hackday = Hackday.find(params[:hackday_id])
+    @project = @hackday.projects.build(filter_params)
+    if @project.save
+      redirect_to hackday_path(@hackday)
+    else
+      @hackday.projects.delete(@project)  # unassociate so it doesn't show in the list?
+      render 'hackdays/show'
+    end
+  end
+
+  def update
+    @project = Project.find(params[:id])
+    @hackday = @project.hackday
+    votes = cookies[:"votes_#{@hackday.id}"].to_i || 0
```

Comment:
session is something that is supported by rails and there are various benefits over cookies. For example sessions are encrypted.